tools/deserialize-proto/init.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/stackrox/rox/generated/storage"
+)
+
+var typeRegistry = make(map[string]proto.Unmarshaler)
+
+func init() {
+	for _, v := range []proto.Unmarshaler{
+		&storage.ActiveComponent{},
+		&storage.AdministrationEvent{},
+		&storage.AuthMachineToMachineConfig{},
+		&storage.AuthProvider{},
+		&storage.Blob{},
+		&storage.CloudSource{},
+		&storage.ClusterHealthStatus{},
+		&storage.ClusterCVE{},
+		&storage.ClusterCVEEdge{},
+		&storage.ComplianceConfig{},
```

Comment:
Could this list be derived from the list at `tools/generate-helpers/pg-table-bindings/list.go` so that it doesn't need to be maintained? If the other list doesn't have every proto, perhaps it is a starting point and others are added here.  Wondering too if can do some fancy go reflection to automatically discover all the `storage*` structs.

Reply:
That's actually where this list comes from. Do you know how I can reference that list?

Reply:
Hrm, just realized they are both are in main packages may have to move the 'list' into a separate shared package for that to be re-usable... or some hack by parsing `tools/generate-helpers/pg-table-bindings/list.go` as text, extracting the struct names, and then using reflect to instantiate them maybe?

Reply:
Excellent! TIL - proto lib maintains an internal string/type mapping.